### PR TITLE
Silabs robin master merge 220118 40p

### PR DESCRIPTION
**Pull request by MikeOpenHWGroup**

### CODE REVIEW

File: cv32e40p/tests/uvmt/compliance-tests/uvmt_cv32e40p_firmware_test.sv [hunk 1]
```diff
@@ -34,18 +35,12 @@
...
-   //}
-
    constraint test_type_cons {
      test_cfg.tpt == PREEXISTING_SELFCHECKING;
    }
 
-
-   `uvm_component_utils(uvmt_cv32e40p_firmware_test_c)
+   `uvm_component_utils_begin(uvmt_cv32e40p_firmware_test_c)
+   `uvm_object_utils_end
```
Comment (silabs-hfegran) [inline cv32e40p/tests/uvmt/compliance-tests/uvmt_cv32e40p_firmware_test.sv:43]: out of curiosity, why was this changed when there is nothing included in the begin-end statement?

File: lib/uvm_agents/uvma_obi_memory/src/comps/uvma_obi_memory_drv.sv [hunk 2]
```diff
@@ -404,12 +376,22 @@ task uvma_obi_memory_drv_c::drv_mstr_read_req(ref uvma_obi_memory_mstr_seq_item_
...
+   end
+
    // Wait for grant
    while (mstr_mp.drv_mstr_cb.gnt !== 1'b1) begin
       @(mstr_mp.drv_mstr_cb);
    end
-   
+
    // Wait for rvalid
    if (mstr_mp.drv_mstr_cb.rvalid !== 1'b1) begin
```
Comment (silabs-hfegran) [inline lib/uvm_agents/uvma_obi_memory/src/comps/uvma_obi_memory_drv.sv:396]: Should this if be here?

File: cv32e40p/tests/uvmt/compliance-tests/uvmt_cv32e40p_firmware_test.sv (hunk 1 above)
Comment (MikeOpenHWGroup) [reply cv32e40p/tests/uvmt/compliance-tests/uvmt_cv32e40p_firmware_test.sv:43]: No reason, other than my mild case of OCD.  I dislike the ability to have a naked `uvm_component_utils_begin()` macro without a trailing `uvm_object_utils_end`.

**Comment by MikeOpenHWGroup** [comment]
@all, this PR is not intended to be merged - it was done as an aid to help me with the review of @silabs-robin's `silabs-robin_master_merge_220118_40p` branch.   Apologies for not making that clear in the description.

However, I'd like to keep this PR open until @silabs-hfegran and I resolve his question about `lib/uvm_agents/uvma_obi_memory/src/comps/uvma_obi_memory_drv.sv`, above.

File: cv32e40p/tests/uvmt/compliance-tests/uvmt_cv32e40p_firmware_test.sv (hunk 1 above)
Comment (silabs-hfegran) [reply cv32e40p/tests/uvmt/compliance-tests/uvmt_cv32e40p_firmware_test.sv:43]: Fair point, though I presume you meant the naked `uvm_component_utils()`, as far as I know that i just a substitution for the ones you replaced it with, to be used when there are no actual contents to go between the begin and end-macros.

File: lib/uvm_agents/uvma_obi_memory/src/comps/uvma_obi_memory_drv.sv (hunk 2 above)
Comment (silabs-hfegran) [reply lib/uvm_agents/uvma_obi_memory/src/comps/uvma_obi_memory_drv.sv:396]: Sorry about this, it's not part of the diff, but github was playing some tricks on me when I hid whitespace changes - my slight OCD trigger when there is a section of code that states the following (superfluous if):
```
if (!condition) begin
   while (!condition) begin
      (...)
   end
end
```

**Comment by silabs-robin** [comment]
FIY, here is the order I do these merges in:

Branched off latest `master` into `silabs-robin/pr/master_merge_220118` (this is the branch I intend to use for a PR later).
Branched off that new branch, into `silabs-robin/pr/master_merge_220118_40p` (the one used in this "fake" merge), and here I merged `40p_release` into.
Branched off that branch into `silabs-robin/pr/master_merge_220118_40p_40x`, where I additionally merged in `40x_release` on top of where I already merged `40p_release`.
Same with `silabs-robin/pr/master_merge_220118_40p_40x_40s`.
Finally, `silabs-robin/pr/master_merge_220118_40p_40x_40s_x2s` is where I am currently doing the manual merge/copy of the 40x directory into the 40s directory.
When that `...40p_40x_40s_x2s` branch is done (and passing "rel_check"), I'll merge that all into my `silabs-robin/pr/master_merge_220118` and make a PR into openhwgroup's `master`.

File: lib/uvm_agents/uvma_obi_memory/src/comps/uvma_obi_memory_drv.sv (hunk 2 above)
Comment (MikeOpenHWGroup) [reply lib/uvm_agents/uvma_obi_memory/src/comps/uvma_obi_memory_drv.sv:396]: Nice catch!   Thanks for this.  I will create a PR to get this fixed.

**Comment by MikeOpenHWGroup** [comment]
OK, this PR has fulfilled its purpose and can be closed.   Thanks for the comments.

@silabs-robin, your summary of "the order I do these merges in" is worth preserving and you already have a [PR](https://github.com/openhwgroup/core-v-docs/pull/497) assigned to you for that purpose.  No good deed goes unpunished.
